src/initial-submission/components/FileDetailsForm.tsx

```diff
@@ -21,6 +25,45 @@ const FileDetailsForm = ({ initialValues = { id: '' } }: Props): JSX.Element =>
     });
 
     const [saveCallback] = useMutation(saveFilesPageMutation);
+
+    const [uploadManuscriptFile] = useMutation(uploadManuscriptMutation);
+
+    // this might be better placed in its own hook or wrapper component so changes don't cause whole page rerender
+    const [manuscriptStatus, setManuscriptStatus] = useState<{ fileStored?: {}; uploadInProgress?: {} }>({});
+
+    useEffect(() => {
+        if (initialValues.manuscriptFile) {
+            setManuscriptStatus({
+                fileStored: {
+                    fileName: initialValues.manuscriptFile.filename,
+                    previewLink: initialValues.manuscriptFile.url,
+                },
+            });
+        }
+    }, []);
+
+    const onManuscriptUpload = (files: File[]): void => {
+        setManuscriptStatus({
+            fileStored: manuscriptStatus.fileStored,
+            uploadInProgress: {
+                progress: 0,
+                fileName: files[0].name,
+            },
+        });
+
+        uploadManuscriptFile({
+            variables: {
+                id: initialValues.id,
+                file: files[0],
+                fileSize: files[0].size,
+            },
+        }).then(data => {
+            setManuscriptStatus({
+                fileStored: data.data.uploadManuscript.manuscriptFile,
```

Comment:
Tests throw an error at this point when resolving the promise. 
```TypeError: Cannot read property 'manuscriptFile' of undefined```

doing a console.log of `data` outputs this: 
```{ data: { data: { uploadManuscript: [Object] } } }```

and JSON.stringifying `data` outputs:
```{"data":{"data":{"uploadManuscript":{"manuscriptFile":{"url":"http://localhost/file.pdf","filename":"testfile.pdf"}}}}}```

I'm unsure why its serialising the uploadManuscript object instead of just treating it like JSON as it's all being instantiated in the test file as an object literal. It looks like the cause of the error being thrown though even though we can see the properties are on the Object from JSON.stringifying it.

Calling `debug()` in the test after the `await wait()`   if we hard code line 62 here `data.data.uploadManuscript.manuscriptFile` to be `{ url: 'http://localhost/file.pdf', filename: 'testfile.pdf' }` shows what we'd expect and the tests pass without errors so I'm unsure what's happening.

Reply:
Doh! named the param data too so it should have been data.data.data 🥇